doc/format.md

```diff
@@ -317,6 +317,7 @@ This could also be an obfuscated string, for example a stack string.
 
 The parameter is a string describing the string.
 This can be the verbatim value, or a regex matching the string.
+Verbatim values containing special characters must be surrounded by double quotes and each special character must be escaped.
```

Comment:
i think we need more detail here. what is a special character? lets enumerate those now before we forget.
```suggestion
Verbatim values containing special characters must be surrounded by double quotes and each special character must be escaped.
A special character is one of:
  - a backslash, which should be represented as `string: "\\"`
  - a newline or other non-space whitespace (e.g. tab, CR, LF, etc), which should be represented like `string: "\n"`
  - a double quote, which should be represented as `string: "\""`
  - a leading slash (not for a regex), which should be represented as `string: "/"`
```

Reply:
what about `:` and `#`?

Reply:
what about spaces, see comment below?

Reply:
I think the current list is comprehensive now that we enforce all strings must start/end with double quotes.